admin_guide/scheduling/scheduler.adoc

```diff
@@ -657,22 +652,20 @@ regardless of its value.
 "predicates":[
       {
          "name":"<name>", <1>
-         "weight" : "1" <2>
-         "argument":{
+          "argument":{
             "labelsPresence":{
                "labels":[
-                  "<label>" <3>
-                presence: true/false <4>
+                  "<label>" <2>
+                presence: true/false <3>
```

Comment:
the above part seems completely incorrect. We are talking about priorities and label preference but the above shows predicate and label presence. The above should be fixed.

Reply:
@aveshagarwal I changed these parameters back to how they were in previous version, before the changes to the scheduling topic were made. 
Did you see  @jtudelag comments in the BZ: [_scheduling based on custom labels does not work... Most of the examples then do not apply...]( https://bugzilla.redhat.com/show_bug.cgi?id=1596472#c1). I am not sure what needs to be changed.

Reply:
@mburke5678, I guess what @aveshagarwal means is that first of all, this is a priority function, not a predicate. Then it is "LabelPreference", not "LabelsPresence".

Also, as a function, It always should have a weight. It is still correct in OCP 3.5 examples:
https://docs.openshift.com/container-platform/3.5/admin_guide/scheduler.html#configurable-priority-functions

Reply:
So, now I have Configurable Priority: `LabelPreference`, with `weight`, and `label` singlular/
Configurable Predicate: `LabelsPresence`, no weight and `labels` plural. Also has `presence: true/false`.
Need a definition for `LabelPreference`. They are slightly different
_LabelPreference prefers nodes that have a particular label defined or not, regardless of its value._
_LabelsPresence checks whether a particular node has a certain label defined or not, regardless of its value._

Reply:
> @mburke5678, I guess what @aveshagarwal means is that first of all, this is a priority function, not a predicate. Then it is "LabelPreference", not "LabelsPresence".

> Also, as a function, It always should have a weight. It is still correct in OCP 3.5 examples:
https://docs.openshift.com/container-platform/3.5/admin_guide/scheduler.html#configurable-priority-functions

+1

> Main question is if this is still honoured by the scheduler in OCP. 

If I understand your questions correctly so yes LabelPreference and LabelPresence still work.

> Also, the explanation of what the LabelPreference function does should be re-written, as now it is just a copy paste of the LabelPresence predicate. @aveshagarwal should answer these questions, I dont have enought knowledge to do it ;(.

+1

Reply:
@aveshagarwal 
"If true, higher priority is given to nodes that have the label"

What is "true" in this case? The presence of the specified label on a node?

`**_LabelPreference_** give priority based on the specified label. 
If the label is present on a node, that node is given priority.
If no label is specified, priority is given to nodes that do not have a label.`

Reply:
> What is "true" in this case? The presence of the specified label on a node?

Yes.

Reply:
presence should be quoted